### DISCUSSION TITLE
Why thanos ruler needs to send TSDB blocks to remote storage

### DISCUSSION THREAD

**Discussion by dmilind** [thanos-io/thanos#3996] [Questions & Answers] [answered]
The receiver pushes TSDB blocks to remote storage already, why do we need ruler to ship TSDB blocks to remote location. How ruler shipped block is different from receiver shipped block. 
Is there any option in ruler to stop shipping TSDB blocks to remote storage ?

**Comment by yeya24** [comment, marked as answer]
> Is there any option in ruler to stop shipping TSDB blocks to remote storage ?
Yes, you can definitely do that. You can just not add the object store configuration for ruler.

The TSDB blocks generated from receivers contain samples sent by other clients via remote write. And the TSDB blocks generated by the ruler contain samples generated from the ruler itself.
If you have alerting rules, the ruler creates two metrics `ALERTS` and `ALERTS_FOR_STATE` for the current alert status. If you have recording rules, the ruler evaluates the recording rules and generates new metrics. So that's what its TSDB blocks for.

So if you want to persist the metrics generated by the ruler for a longer time, you have to upload them to the remote storage.

But you are right, we are currently working on a stateless ruler which leverages the Thanos receiver for Ruler storage. Please take a look at https://thanos.io/tip/proposals/202005_scalable-rule-storage.md/.

**Reply by dmilind** [reply]
Thanks @yeya24.
